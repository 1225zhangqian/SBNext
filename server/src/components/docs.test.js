import React from 'react';

import loadEntries from '../entries/load';
import withDocs, { entries } from './docs';

jest.mock('../entries/load');

describe('withDocs', () => {
  test('exports HOC withDocs as default', () => {
    expect(withDocs).toBeDefined();
  });

  test('withDocs should add `docs` property to getInitialProps', async () => {
    const expected = [{ data: {}, content: '' }];
    const Component = withDocs(({ docs }) => <div>There are {docs.length} docs</div>);

    loadEntries.mockReturnValueOnce(expected);
    const actual = await Component.getInitialProps();

    expect(actual.docs).toMatchSnapshot();
  });
});

describe('entries', () => {
  test('exports entries function', () => {
    expect(entries).toBeDefined();
  });
});
