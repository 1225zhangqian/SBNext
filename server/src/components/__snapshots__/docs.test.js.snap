// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withDocs withDocs should add \`docs\` property to getInitialProps 1`] = `
Array [
  Object {
    "content": "",
    "data": Object {},
  },
]
`;
